### ISSUE TITLE
[BUG] Cellpose model fails on parts of image

### ISSUE THREAD

**Issue by ingvildeb** [MouseLand/cellpose#1276]
**Describe the bug**
I have trained a Cellpose 4 model on chunks (128*128 px) from brain images. When I apply the model to the full-size images, it generally works well. However, it fails completely on seemingly random parts of the images, which look rectangular (reminiscent  of the chunk shapes). 

I normalized my images before labelling, training and applying the model. The model was trained using recommended settings:

python -m cellpose --train --dir Z:\Labmembers\Ingvild\Cellpose\Aldh_model\training_chunks --test_dir Z:\Labmembers\Ingvild\Cellpose\Aldh_model\test_data --learning_rate 0.00001 --weight_decay 0.1 --n_epochs 100 --train_batch_size 1 --use_gpu --model_name 071825_aldh_cspam_100epoch --verbose --mask_filter _seg.npy

And applied to a whole image via the command line. Applying the model on the whole image in the GUI gives the same result.

I could not think of any good reason it should fail on the particular part of the image, as it looked fairly similar to other parts of the image. After training on several hundred chunks from different images, I also never had the model perform so poorly on any single chunk. I therefore cropped out the part of the image where this was happening and ran the model on only that crop, and in this case it works..

**Screenshots**
Brain image and resulting segmentation side-by-side, area where model underperforms circled:
<img width="1264" height="791" alt="Image" src="https://github.com/user-attachments/assets/5524bc0e-3376-4a1a-8660-970a6ad35137" />

Cropped view of the area ran in the Cellpose GUI with the same model:

<img width="1207" height="1024" alt="Image" src="https://github.com/user-attachments/assets/71827c84-afc0-4ea7-804e-d9b205ac6222" /> 

<img width="1205" height="1027" alt="Image" src="https://github.com/user-attachments/assets/74a71a88-5c6c-4835-a562-449d00fde67c" />

I'm not sure if there's something obvious I am missing or if this is a bug, but would appreciate any help on this.

**Comment by mrariden** [comment]
It's hard to say but could be related to the tiling steps as the image is passed into the net. There is a normalization that is done on the images at this step when you could try turning off since you're pre-normalizeing. It's `--no_norm` in the CLI

**Comment by mrariden** [comment]
@arthurgodino That's a very extreme example. Can you upload a sample that produces this kind of behavior? It will make the debugging a lot easier with this clear of an issue.

**Comment by mrariden** [comment]
@codyslater Are you using any denoising or pre-processing on your image?

**Comment by codyslater** [comment]
@mrariden no at the moment there is no additional processing happening to the image after the basic stitching of tiles at the microscope.

**Comment by mrariden** [comment]
What we think is happening now is that the network is unsure about the size of objects. It makes sense that putting smaller tiles through the network would cue the network that the objects are smaller since there is less larger/long range information in the tiles. 

We're still working on a comprehensive solution.

**Comment by m-albert** [comment]
Thank you for looking into this! I can confirm the behaviour for an input image of size ~4000x5000, occurring independently of preprocessing.

<img width="1085" height="736" alt="Image" src="https://github.com/user-attachments/assets/bc0ef8fb-4e28-479d-83e7-1d2846de6aef" />

**Comment by DarioS** [comment]
I can share raw input data if useful. I use the pretrained model with `diameter = 15` and `cellprob_threshold = -2.0`.

<img width="1209" height="594" alt="Image" src="https://github.com/user-attachments/assets/b194d39a-6ea2-4ec9-8284-0878c88c8f2b" />

Imaging mass cytometry is sadly 1 µm per pixel. The custom parameters helped on some images but not others.

**Comment by Umaarasu** [comment]
@mrariden @m-albert ..Hi, have you solved this issue? Because I am facing a similar one with Cellpose3 when trying to segment big cells. Could this be a bug issue that can be fixed? because the same pipeline was working fine for the same image and suddenly its throwing error. 

<img width="1226" height="704" alt="Image" src="https://github.com/user-attachments/assets/12deffa3-0f0e-4a06-9362-35dc6e925f9d" />

**Comment by linaschmidt** [comment]
Is there any news on this? It keeps happening for most of my cpsam segmentations.

**Comment by DarioS** [comment]
It might be a fundamental limitation of the model, or it would surely have been fixed in the past year. Have you tried [SimpleSeg](https://bioconductor.org/packages/release/bioc/html/simpleSeg.html), which does not require input image tiling to make the input a fixed dimension, which A.I. models sadly all seem to require?

**Comment by Zailushang211** [comment]
Hi all, After days of exploration, I found that it is not a problem of the model. 
It happens because one of the image channels is empty,  all zero pixel,  just convert your image to gray, then input to the model.

**Comment by carsen-stringer** [comment]
Hi everyone, for the missing tiles, we have resolved this issue mostly by retraining the models without using very low contrast levels in the augmentations. Please try the new `cpsam_v2` and `cpdino` and `cpdino-vitb` models, `pip install git+https://github.com/mouseland/cellpose`

I will close this issue but let us know if you still have issues

**Comment by DarioS** [comment]
Issue still persists but is no longer obvious square tiles. I tried `pretrained_model='cpsam_v2'`.
<img width="1620" height="937" alt="Image" src="https://github.com/user-attachments/assets/17faf42e-20c7-4523-90b1-db0f63f79b37" />
Also, the cell masks often appear to be in dark spots, rather than surrounding any measured signal.
<img width="471" height="869" alt="Image" src="https://github.com/user-attachments/assets/ccf18aee-2e99-42bf-8472-3ec27d2e79d1" />
Interested to know the result for the more traditional assays which others have previously shared.
